lib/core/lib/ros/scheduler/tenant_handler.rb

```diff
@@ -44,6 +44,7 @@ def per_tenant
           Tenant.find_each do |tenant|
             next if job_enqueued(tenant)
 
+            sleep(1)
```

Comment:
This is very racy and will simply paper over the issue in most cases but still trigger when you least expect it.